### PR TITLE
Product Collection: Support registering product collection that isn't displayed in Collection Chooser

### LINKED ISSUE #50762: Product Collection: Support registering product collection that isn't displayed in inserter
### Describe the solution you'd like

I'd like a way to register a product collection that isn't meant to be inserted by store owners. My intent is to use it internally to output custom collections (probably as an inner block in the add to cart with option block.

image

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by imanish003**
### Changes proposed in this Pull Request:

Closes #50762 

This PR enhances product collection registration by using `scope` argument for controlling visibility.

1. Collections with `scope: ['block', 'inserter']` (default):
   - Visible in the Collection Chooser
   - Accessible via the "Choose Collection" toolbar button

2. Collections with `scope: ['inserter']`:
   - Hidden from the Collection Chooser
   - "Choose Collection" toolbar button will not be visible
   - Visible in the block inserter

3. Collections with `scope: ['block']`:
   - Visible in the Collection Chooser
   - "Choose Collection" toolbar button will be visible
   - Not visible in the block inserter

4. Collections with `scope: []`:
   - Hidden from the Collection Chooser
   - "Choose Collection" toolbar button will not be visible
   - Not visible in the block inserter

In short, `block` scope means that the collection will be shown in Collection Chooser and "Choose Collection" toolbar button will be visible.

Related Slack discussion: p1724747359216099-slack-C05UMTA7W0P

### How to test the changes in this Pull Request:

**Prerequisite:** Download and install following plugin: [product-collection-testing.zip](https://github.com/user-attachments/files/17411745/product-collection-testing.zip). This plugin contains 3 custom collections named "My Custom Collection - with block scope", "My Custom Collection - with inserter scope", and "My Custom Collection - without scope argument".

**Case 1: Collection has scope: ['block']**

1. Create a new post or page and add the Product Collection block.
2. Verify that in collection chooser, you can see collection with name "My Custom Collection - with block scope"
	- Sidenote: This collection has scope: ['block']
3. Click on "My Custom Collection - with block scope" to select it. 
4. Verify that "Choose Collection" toolbar button is visible

**Case 2: Collection has scope: ['inserter']**	

1. Create a new post or page and add the Product Collection block.
2. Verify that in collection chooser, you can't see collection with name "My Custom Collection - with inserter scope"
3. Verify that in inserter, you can see collection with name "My Custom Collection - with inserter scope"
4. Add "My Custom Collection - with inserter scope" to the post from inserter.
5. Verify that "Choose Collection" toolbar button is not visible

**Case 3: Collection has no scope argument**
_Sidenote_: When scope argument is not defined, default scope is ['block', 'inserter']

1. Create a new post or page and add the Product Collection block.
2. Verify that in collection chooser, you can see collection with name "My Custom Collection - without scope argument"
3. Verify that in inserter, you can see collection with name "My Custom Collection - without scope argument"
4. Add "My Custom Collection - without scope argument" to the post from inserter.
5. Verify that "Choose Collection" toolbar button is visible

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [x] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Product Collection: Support registering product collection that isn't displayed in Collection Chooser

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/toolbar-controls/index.tsx [hunk 1]
```diff
@@ -24,11 +25,19 @@ export default function ToolbarControls(
 		[ props ]
 	);
 
+	const collection = getCollectionByName( props.attributes.collection );
+	const isShowCollectionChooserToolbar =
```
Comment (kmanijak) [inline plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/toolbar-controls/index.tsx]: Sorry, I don't understand this variable name. Did you mean `showCollectionChooserToolbar`?

**Review by kmanijak — COMMENTED** [review]
I left one minor comment regarding variable name. One thing I'd request is to update documentation regarding `scope` variable. Because we're aligned with Variations API but it's a custom "interpretation" or "implementation" of `scope: [ 'block' ]` (it occurs a bit differently in Variations).

File: plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/toolbar-controls/index.tsx (hunk 1 above)
Comment (imanish003) [reply plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/toolbar-controls/index.tsx]: Yes, Made the changes in [02ca2de](https://github.com/woocommerce/woocommerce/pull/52057/commits/02ca2de245e8f85551716d5dc642e75bd7afd1d5)

**Comment by imanish003** [comment]
Hey @kmanijak, This PR is ready for review again:
- Added E2E tests
- Updated documentation

**Review by kmanijak — APPROVED** [review]
All looks good and tests well! Thanks for adding E2E tests as well! Great work!

Could you just update the PR description a bit? Currently it says:

> 1. Collections with scope: ['block', 'inserter'] (default):
> ...
> 2. Collections with scope: ['inserter']:
> ...
> 3. Collections with empty scope or without 'block':

2 and 3 overlap. I think we should explicitly describe remaining cases (`['block]` and `[]`):

1. Collections with scope: ['block', 'inserter'] (default)
2. Collections with scope: ['inserter']
3. Collections with scope: ['block']
4. Collections with scope: []

**Comment by imanish003** [comment]
> 2 and 3 overlap. I think we should explicitly describe remaining cases (`['block]` and `[]`):
> 
> 1. Collections with scope: ['block', 'inserter'] (default)
> 2. Collections with scope: ['inserter']
> 3. Collections with scope: ['block']
> 4. Collections with scope: []

@kmanijak Good point! I have made the changes to PR description 🤝

**Comment by xristos3490** [comment]
Thanks for working on this, @imanish003! Now that it's merged, I'll initiate the work to update our premium offerings in the Product Recommendations extension!

**Pull request merged**